SK2 alpha9 regression

```
2017-10-07 05:40:18|ArchiSteamFarm-9864|INFO|nany40|OnDisconnected() Disconnected from Steam!
2017-10-07 05:40:18|ArchiSteamFarm-9864|INFO|nany40|OnDisconnected() Reconnecting...
2017-10-07 05:40:18|ArchiSteamFarm-9864|INFO|nany40|Connect() Connecting...
2017-10-07 05:40:18|ArchiSteamFarm-9864|FATAL|ASF|OnUnhandledException() System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SteamKit2.Internal.CMClient.Disconnect()
   at SteamKit2.Internal.CMClient.Connect(ServerRecord cmServer)
   at ArchiSteamFarm.Bot.<Connect>d__104.MoveNext() in C:\projects\archisteamfarm\ArchiSteamFarm\Bot.cs:line 1006
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ArchiSteamFarm.Bot.<OnDisconnected>d__137.MoveNext() in C:\projects\archisteamfarm\ArchiSteamFarm\Bot.cs:line 1623
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

This looks like SK2 internals. As you can see caused by reconnect attempt in `OnDisconnected()`. Thanks.

Ref: https://github.com/JustArchi/ArchiSteamFarm/issues/665

Haha, it _is_ related to #455. Looks like a strange consequence of `TaskContinuationOptions.OnlyOnFaulted`.

```
Microsoft (R) Visual C# Interactive Compiler version 2.3.2.62116
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
> using System.Threading.Tasks;
> Task.FromResult(true)
(1,1): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
Task<bool>(Id = 35, Status = RanToCompletion, Method = "{null}", Result = "True") { AsyncState=null, CancellationPending=false, CreationOptions=None, Exception=null, Id=35, Result=true, Status=RanToCompletion }
> Task.FromResult(true).ContinueWith(t => { }, TaskContinuationOptions.OnlyOnFaulted)
(1,1): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
ContinuationTaskFromResultTask<bool>(Id = 54, Status = Canceled, Method = "{null}") { AsyncState=null, CancellationPending=false, CreationOptions=None, Exception=null, Id=54, Status=Canceled }
```

--- a/SteamKit2/SteamKit2/Steam/CMClient.cs
+++ b/SteamKit2/SteamKit2/Steam/CMClient.cs
@@ -175,7 +175,7 @@ namespace SteamKit2.Internal
                 connection.NetMsgReceived += NetMsgReceived;
                 connection.Connected += Connected;
                 connection.Disconnected += Disconnected;
-                connection.Connect( record.EndPoint, (int)ConnectionTimeout.TotalMilliseconds );
+                connection.Connect( record.EndPoint, ( int )ConnectionTimeout.TotalMilliseconds );
             }, TaskContinuationOptions.ExecuteSynchronously).ContinueWith(t =>
             {
                 if ( t.IsFaulted )

--- a/SteamKit2/SteamKit2/Steam/CMClient.cs
+++ b/SteamKit2/SteamKit2/Steam/CMClient.cs
@@ -176,14 +176,14 @@ namespace SteamKit2.Internal
                 connection.Connected += Connected;
                 connection.Disconnected += Disconnected;
                 connection.Connect( record.EndPoint, (int)ConnectionTimeout.TotalMilliseconds );
-            }).ContinueWith(t =>
+            }, TaskContinuationOptions.ExecuteSynchronously).ContinueWith(t =>
             {
                 if ( t.IsFaulted )
                 {
                     DebugLog.WriteLine( nameof(CMClient), "Unhandled exception when attempting to connect to Steam: {0}", t.Exception );
                     OnClientDisconnected( userInitiated: false );
                 }
-            });
+            }, TaskContinuationOptions.ExecuteSynchronously);
         }
 
         /// <summary>

--- a/SteamKit2/SteamKit2/Steam/CMClient.cs
+++ b/SteamKit2/SteamKit2/Steam/CMClient.cs
@@ -175,12 +175,15 @@ namespace SteamKit2.Internal
                 connection.NetMsgReceived += NetMsgReceived;
                 connection.Connected += Connected;
                 connection.Disconnected += Disconnected;
-                connection.Connect( record.EndPoint, ( int )ConnectionTimeout.TotalMilliseconds );
-            } ).ContinueWith( t =>
+                connection.Connect( record.EndPoint, (int)ConnectionTimeout.TotalMilliseconds );
+            }).ContinueWith(t =>
             {
-                DebugLog.WriteLine( nameof(CMClient), "Unhandled exception when attempting to connect to Steam: {0}", t.Exception );
-                OnClientDisconnected( userInitiated: false );
-            }, TaskContinuationOptions.OnlyOnFaulted );
+                if ( t.IsFaulted )
+                {
+                    DebugLog.WriteLine( nameof(CMClient), "Unhandled exception when attempting to connect to Steam: {0}", t.Exception );
+                    OnClientDisconnected( userInitiated: false );
+                }
+            });
         }
 
         /// <summary>

